mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -66,8 +71,27 @@ def message_post(
                         subject = rendered_subject_template + subject
                     elif template.position == "append_after":
                         subject += rendered_subject_template
+                    elif template.position == "inside_replace":
+                        rendered_subject_to_replace = self.env[
+                            "mail.template"
+                        ]._render_template(
+                            template_src=template.subject_to_replace,
+                            model=self._name,
+                            res_ids=[self.id],
+                        )[self.id]
+                        if rendered_subject_to_replace:
+                            # To avoid empty string replacements
+                            subject = subject.replace(
+                                rendered_subject_to_replace,
+                                rendered_subject_template,
+                            )
                 except Exception:
-                    rendered_subject_template = False
+                    _logger.warning(
+                        _(
+                            "There is an error with {record_name} in the mail message "
+                            "custom subject {t_name}"
+                        ).format(t_name=template.name, record_name=self.display_name)
+                    )
```

Comment:
Logs messages are not translated:

```suggestion
                    _logger.warning(
                        f"There is an error with {self.display_name} in the mail message "
                        "custom subject {template.name}"
                    )
```